plugins/lookup/validate.py

```diff
@@ -0,0 +1,167 @@
+      documentation that is represented in the value of C(engine) option.
+    - For additional plugin configuration options refer the individual C(validate) plugin
+      documentation that is represented by the value of C(engine) option.
+    - The plugin configuration option can be either passed as I(key=value) pairs within lookup plugin
+      or task or environment variables.
+    - The precedence the C(validate) plugin configurable option is the variable passed within lookup plugin
+      as I(key=value) pairs followed by task variables followed by environment variables.
+"""
+
+EXAMPLES = r"""
+- name: set facts for data and criteria
+  set_fact:
+    data: "{{ lookup('file', './validate/data/show_interfaces_iosxr.json')}}"
+    criteria: "{{ lookup('file', './validate/criteria/jsonschema/show_interfaces_iosxr.json')}}"
+
+- name: validate data in json format using jsonschema with lookup plugin by passing plugin configuration variable as key/value pairs
+  ansible.builtin.set_fact:
+    data_criteria_checks: "{{ lookup(data, criteria, engine='ansible.utils.jsonschema', draft='draft7') }}"
+
+- name: validate data in json format using jsonschema with lookup plugin by passing plugin configuration variable as task variable
+  ansible.builtin.set_fact:
+    data_criteria_checks: "{{ lookup('ansible.utils.validate', data, criteria, engine='ansible.utils.jsonschema', draft='draft7') }}"
+  vars:
+    ansible_validate_jsonschema_draft: draft3
+"""
+
+RETURN = """
+  _raw:
+    description:
+      - If data is valid returns empty list
+      - If data is invalid returns list of errors in data
+"""
+
+from copy import deepcopy
+
+from ansible.errors import AnsibleError
+from ansible.module_utils._text import to_native
+from ansible.plugins.lookup import LookupBase
+from ansible_collections.ansible.utils.plugins.module_utils.validate.base import (
+    load_validator,
+)
+from ansible_collections.ansible.utils.plugins.module_utils.common.utils import (
+    to_list,
+)
+from ansible_collections.ansible.utils.plugins.module_utils.common.argspec_validate import (
+    check_argspec,
+)
+
+try:
+    import yaml
+
+    try:
+        from yaml import CSafeLoader as SafeLoader
+    except ImportError:
+        from yaml import SafeLoader
+    HAS_YAML = True
+except ImportError:
+    HAS_YAML = False
+
+ARGSPEC_CONDITIONALS = {}
+
+
+class LookupModule(LookupBase):
+    def run(self, terms, variables, **kwargs):
+        if len(terms) < 2:
+            raise AnsibleError(
+                "Missing either 'data' or 'criteria' value in lookup input,"
+                "refer ansible.utils.validate lookup plugin documentation for detials"
+            )
+
+        params = {"data": terms[0], "criteria": terms[1]}
+        if kwargs.get("engine"):
+            params.update({"engine": kwargs["engine"]})
+
+        argspec = deepcopy(DOCUMENTATION)
+        argspec_obj = yaml.load(argspec, SafeLoader)
+        argspec_obj["options"].pop("_terms", None)
```

Comment:
I ended up just leaving _terms out of the lookup doc to avoid this

Reply:
Agreed, I had added it at first but missed removing it. Fixed